### PR TITLE
Module for popen/pclose

### PR DESCRIPTION
**Pull request by mattdurak**
POSIX `popen` and `pclose` and slightly different than Win32 `_popen` and `_pclose`, this module hides the differences.

### CODE REVIEW

File: interfaces/tests/pipe_int/pipe_int.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+{
+    TEST_MUTEX_DESTROY(g_testByTest);
+    gballoc_hl_deinit();
+}
+
+TEST_FUNCTION_INITIALIZE(c)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
```
Comment (dcristoloveanu) [inline interfaces/tests/pipe_int/pipe_int.c:46]: > ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework") [](start = 7, length = 65)

Consider one line for acquire and assert

File: interfaces/tests/pipe_int/pipe_int.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    ASSERT_IS_TRUE(feof(file_handle));
+
+    ASSERT_ARE_EQUAL(int, 0, pipe_pclose(file_handle, &exit_code));
+    ASSERT_ARE_EQUAL(int, 0, exit_code);
+}
+
+/*Tests_SRS_PIPE_42_005: [ pipe_pclose shall store the result of the executed command in exit_code. ]*/
+TEST_FUNCTION(pipe_open_close_succeeds_returns_non_zero_exit)
```
Comment (dcristoloveanu) [inline interfaces/tests/pipe_int/pipe_int.c:84]: >_exit [](start = 55, length = 5)

non_zero_exit_code

File: linux/tests/pipe_linux_ut/mock_pipe.c [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <errno.h>
```
Comment (dcristoloveanu) [inline linux/tests/pipe_linux_ut/mock_pipe.c:4]: Suspect this is not needed. Probably iwyu will say so :-)

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

File: linux/tests/pipe_linux_ut/mock_pipe.c (hunk 3 above)
Comment (mattdurak) [reply linux/tests/pipe_linux_ut/mock_pipe.c:4]: Does iwyu run against Linux as well? Will find out soon I guess...

---
In reply to: [562201137](https://github.com/Azure/c-pal/pull/67#discussion_r562201137) [](ancestors = 562201137)

**Pull request merged**